### ISSUE TITLE
Generating structures in a blocky world

### ISSUE THREAD

**Issue by Aghostcat** [Zylann/godot_voxel#703]
Is there an easy customizable way to generate structures out of voxels in blocky terrain like buildings, dungeons, towns, etc?
I've been looking at the generators and not sure if the way trees are made really apply here.

**Comment by MGilleronFJ** [comment]
No, there isn't. It's not always a trivial thing to solve. You have to come up with that. It might be a bit easier to do with a multipass generator (though it has relatively high costs to pay for that convenience). Otherwise you may exploit determinism and use seed/hashing/voronoi noise, or in the worst case for huge things, manage your own thread-safe data structures to "remember" where things should be as the world progressively generate around.
The only bits of docs I wrote about that is here https://voxel-tools.readthedocs.io/en/latest/procedural_generation/#handling-block-boundaries-with-voxel-structures
